rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp

```diff
@@ -27,6 +27,35 @@ namespace rosbag2_cpp
 {
 namespace readers
 {
+namespace details
+{
+std::vector<std::string> resolve_relative_paths(
+  const std::string & base_folder, std::vector<std::string> relative_files, size_t version = 4)
```

Comment:
```suggestion
  const std::string & base_folder, const std::vector<std::string>& relative_files, int version = 4)
```
Can `relative_files` be passed by const ref?
`version` is also an `int` in [BagMetadata](https://github.com/ros2/rosbag2/blob/267702b1fa5c334ad29ca0c7c0aa670b5ea243a0/rosbag2_storage/include/rosbag2_storage/bag_metadata.hpp#L36)

Reply:
I am passing `relative_files` deliberately by copy as I'll modify the copy and return it. It's a style thing. I can also pass by reference and copy the content in a new vector.

Reply:
That's fine, I just wanted to verify that you intended to pass by copy.